### PR TITLE
halborn audit fixes

### PR DESCRIPTION
**Pull request by reddyismav**

### CODE REVIEW

File: contracts/lib/RLPReader.sol [hunk 1]
```diff
@@ -152,49 +232,54 @@ library RLPReader {
...
+        
         else if (byte0 < STRING_LONG_START)
             itemLen = byte0 - STRING_SHORT_START + 1;
+
         else if (byte0 < LIST_SHORT_START) {
             assembly {
                 let byteLen := sub(byte0, 0xb7) // # of bytes the actual length is
                 memPtr := add(memPtr, 1) // skip over the first byte
-
+
```
Comment (jdkanani) [inline contracts/lib/RLPReader.sol]: Remove it? Better code sanity?

**Review by jdkanani — APPROVED** [review]

File: contracts/lib/RLPReader.sol (hunk 1 above)
Comment (reddyismav) [reply contracts/lib/RLPReader.sol]: Yep !

**Comment by jdkanani** [comment]
@reddyismav It's not compiling. Can you check?

**Comment by reddyismav** [comment]
There is an error in the Rlp reader. Checking.

**Review by jdkanani — APPROVED** [review]

**Comment by nachomazzara** [comment]
Hi folks! in order to keep compatibility with multiple projects still using Solidity 0.6  0.7 versions. Can you replace the pragma with `pragma solidity >=0.6.0 <0.8.0;` create an npm version like `fx-portal@0.7.0` and then create the `0.8` version?

**Comment by reddyismav** [comment]
@nachomazzara okay .. will do it. Thanks for bringing this up

**Pull request merged**